### DISCUSSION TITLE
Cannot start nuxt:  Cannot find module 'nuxt'

### DISCUSSION THREAD

**Discussion by christaggart** [nuxt/nuxt#16520] [Questions]
Brand new project w/node 14 (nvm use 14). I also tested w/yarn and node 17 with the same results. 

Following the docs:

How to reproduce:
> npx nuxi init nuxt3-app
> cd nuxt3-app
> npm install
> npm run dev

```
Nuxt CLI v3.0.0-27398533.8edd481          
                                                                      
  > Local:    http://localhost:3000/

 ERROR  Cannot start nuxt:  Cannot find module 'nuxt'                
Require stack:
- /nuxt3-app/index.js
```

This seems.. odd?

**Comment by wobsoriano** [comment]
Try to remove `node_modules` and lockfile then run `npm install` again

**Reply by christaggart** [reply]
That was the first thing I did, obviously. No dice.

**Reply by wobsoriano** [reply]
`npx nuxi info` output then

**Reply by christaggart** [reply]
Nuxt CLI v3.0.0-27398533.8edd481
- Operating System: Darwin
- Node Version:     v14.17.6
- Nuxt Version:     0.10.0-27192162.13abec6
- Package Manager:  yarn@1.22.17
- Bundler:          Webpack
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -

**Reply by danielroe** [reply]
I think you must have an outdated version of nuxi installed locally. Your Nuxt version should be at least 3.

**Reply by christaggart** [reply]
Yes, there was an old globally installed beta of nuxt3. Thanks.

**Reply by 0xinhua** [reply]
the same error in node v 14.16.0 @christaggart Have you solved the problem yet?

**Comment by antoninbeaufort** [comment]
Same problem here with 

```
Nuxt CLI v3.0.0-27404037.29171bd                                                                                                                                                          09:55:43
RootDir: /home/antoninbeaufort/dev/truc/frontend                                                                                                                                09:55:43
Nuxt project info:                                                                                                                                                                        09:55:43

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.13.2`
- Nuxt Version:     `2.16.0-27358576.777a4b7f`
- Package Manager:  `yarn@3.1.1`
- Bundler:          `Webpack`
- User Config:      `target`, `env`, `head`, `plugins`, `buildModules`, `modules`, `alias`, `generate`, `bridge`
- Runtime Modules:  `@nuxtjs/sentry@5.1.7`
- Build Modules:    `@nuxtjs/eslint-module@3.0.2`, `@nuxtjs/vuetify@1.12.3`, `@nuxtjs/dotenv@1.4.1`, `@nuxt/bridge@0.10.1`
------------------------------
```

**Comment by reqwire** [comment]
For anyone with this problem - do the following to fix:

> 1. Update `package.json`
> 
> ```diff
> {
>   "devDependencies": {
> -    "nuxt3": "latest"
> +    "nuxt": "npm:nuxt3@latest"
>   }
> }
> ```
> 
> 2. Update `nuxt.config`:
> 
> ```diff
> - import { defineNuxtConfig } from 'nuxt3'
> + import { defineNuxtConfig } from 'nuxt'
> 
> export default defineNuxtConfig({ ... })
> ```

(optional, if it still isn't working):

+ Delete `node_modules` folder
+ Delete `package.lock.json` if present
+ Delete `.nuxt` and `.output` folders
+ Rebuild by running `npm install` or `yarn`

**Reply by evildvl** [reply]
Thank you, it's working solution

**Reply by Extarys** [reply]
Thank you so much! One day last week it stopped working and I didn't know what it was :joy: :kissing_heart:

**Reply by lacorde** [reply]
Can't thank you enough !

**Reply by ahsan-a** [reply]
This is it, thank you so much! Just redownload your node modules after that

**Reply by pimboden** [reply]
Thanks! That worked!

**Reply by Diego-Gnoatto** [reply]
thanks it worked here

**Comment by prstwo** [comment]
Here is my solution, it works for me:

- [ ] update package.json
 `  "scripts": {
    "build": "nuxt build",
    "dev": "nuxt dev",
    "generate": "nuxt generate",
    "preview": "nuxt preview"
  },`
` "devDependencies": {
    "nuxt": "3.0.0-rc.1",
  },`
- [ ] update nuxt.config.ts
`import { defineNuxtConfig } from 'nuxt';
`
- [ ] remove node madules and package.lock.json 
- [ ] run: npm install

**Reply by igorcafe** [reply]
Mine is already looking like that...
Which is weird because the dev server runs

![image](https://user-images.githubusercontent.com/85039990/166554998-e3d0a7e1-d742-49f1-9f97-9f458359643f.png)

**Reply by cessarmando93** [reply]
I have the same problem

**Reply by jdalrymple** [reply]
I had a similar problem, but fixed it by adding:

```
nodeLinker: node-modules
```

to my yarn.yml file since im using yarn3 and apparently nuxt doesnt support pnp module linking

**Reply by Zaklo** [reply]
Works well for me, thanks !!

**Reply by lacorde** [reply]
As mentioned here [#5160](https://github.com/nuxt/framework/discussions/5160#discussioncomment-2832704), don't forget to delete `.nuxt `and `.output` folders before running npm/yarn install again !

**Reply by vicventum** [reply]
I have the same problem, I use npm, is there any similar for the `nodeLinker: node-modules` solution but in npm? I am a little junior, sorry if the answer is obvious.

**Comment by szvest** [comment]
Install `@nuxt/kit` as a dev dependency.

**Reply by MartinX3** [reply]
It appears while using vue > 2.7.5
back to 2.7.5 everything is fine without `@nuxt/kit`

Removing the `package-lock.json` file and the `node_modules` folder fixed this issue without adding `@nuxt/kit`

**Comment by jjjjose** [comment]
I have the same problem
![Captura de Pantalla 2022-08-11 a la(s) 00 39 46](https://user-images.githubusercontent.com/5414128/184065781-d743171e-0278-4df6-aa9b-26bd85c56ac6.png)

**Reply by danielroe** [reply]
Would you provide more info/your code/a reproduction?

**Reply by jjjjose** [reply]
just use create a new project with
npx nuxi init nuxt-app
install yarn install
nothing else

**Reply by danielroe** [reply]
Thanks! Found something... I'm guessing you have a `.` in your directory name?

**Reply by scherii** [reply]
Yes, that was it for me. Same problem and the directory name is a domain (with a `.`).

Somehow another project with the same structure is not affected. Difference between the two is that the affected one has 2 dots (`subdomain.domain.tld`) and the other just one.

**Reply by jjjjose** [reply]
> Thanks! Found something... I'm guessing you have a `.` in your directory name?

exactly yesterday create another project without dot in the name
and gave normal
I use points to order the project, is there any other way to solve this problem?

**Reply by scherii** [reply]
@jjjjose Have you already tried to update to [Release Candidate 7](https://github.com/nuxt/framework/releases/tag/v3.0.0-rc.7)?
I have not yet been able to verify if the problem persists, but will let you know if I find out more.

✍️ Edit:
Okay, I did a quick test with a directory with one and with two dots in it and it looks quite like it now works as expected with RC7. Will try it on the real project later today.

**Reply by jjjjose** [reply]
indeed, the rc7 version corrects this error

**Reply by szulcus** [reply]
> Thanks! Found something... I'm guessing you have a `.` in your directory name?

I've had a similar problem for a long time. In my case, I think the "#", which I mark my own projects with, was disturbing. Any chance you can fix it someday?

**Comment by Tonysmark** [comment]
Should it be th problem that  yarn PnP is  unavalliable? Cause I got this same error using yarn 3

```
Nuxt CLI v3.0.0-rc.7                                                                                16:40:00
RootDir: E:/Project/nuxt-app                                                                        16:40:00
Nuxt project info:                                                                                  16:40:00

------------------------------
- Operating System: `Windows_NT`
- Node Version:     `v14.19.0`
- Nuxt Version:     `npm:nuxt3@latest`
- Package Manager:  `yarn@3.2.1`
- Builder:          `webpack`
- User Config:      `-`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

```

package.json

```json
{
  "private": true,
  "scripts": {
    "build": "nuxt build",
    "dev": "nuxt dev",
    "generate": "nuxt generate",
    "preview": "nuxt preview"
  },
  "devDependencies": {
    "nuxt": "npm:nuxt3@latest"
  },
  "dependencies": {
    "@nuxt/kit": "^3.0.0-rc.7"
  }
}
```

**Reply by sachinkmohan** [reply]
nuxt was not installed for me. So I added nuxt to dependencies, removed `node_modules & package-lock.json` and did `npm i`. Started working again.
